src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java

```diff
@@ -429,7 +429,11 @@ public CreateAnomalyDetectorTool create(Map<String, Object> map) {
                 throw new IllegalArgumentException("model_id cannot be empty.");
             }
             String modelType = (String) map.getOrDefault("model_type", ModelType.CLAUDE.toString());
-            if (!ModelType.OPENAI.toString().equalsIgnoreCase(modelType) && !ModelType.CLAUDE.toString().equalsIgnoreCase(modelType)) {
+            // if model type is empty, use the default value
+            if (modelType.isEmpty()) {
```

Comment:
Not understand why this change as in line 431 there's already a default value which should suggest `modelType` is not empty, isn't it?

Reply:
No, the default value takes effect only if the key `model_type` doesn't exist in the map or its value is null.

Reply:
Got it, this is to handle empty string case instead of null value.